rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java

```diff
@@ -99,19 +103,35 @@ public static void main(String[] args) throws Exception {
     public static class EnrichmentFunction
             extends RichCoFlatMapFunction<TaxiRide, TaxiFare, RideAndFare> {
 
+        private ValueState<RideAndFare> rideAndFare;
+
         @Override
-        public void open(Configuration config) throws Exception {
-            throw new MissingSolutionException();
+        public void open(Configuration config) {
+            rideAndFare = getRuntimeContext().getState(new ValueStateDescriptor<>("rideAndFare", RideAndFare.class));
         }
 
         @Override
         public void flatMap1(TaxiRide ride, Collector<RideAndFare> out) throws Exception {
-            throw new MissingSolutionException();
+            if (rideAndFare.value() == null) {
+                RideAndFare initialRideAndFare = new RideAndFare(ride, null);
+                rideAndFare.update(initialRideAndFare);
+            } else {
+                RideAndFare rideAndFareForCollect = rideAndFare.value();
+                rideAndFareForCollect.setRide(ride);
+                out.collect(rideAndFareForCollect);
+            }
```

Comment:
Remember to clear the state once it is not needed anymore. In a streaming application with an unbounded input, the state will also grow indefinitely unless it is managed.

Reply:
done